coderd/util/slice/slice.go

```diff
@@ -62,6 +62,20 @@ func Overlap[T comparable](a []T, b []T) bool {
 	})
 }
 
+func UniqueFunc[T any](a []T, equal func(a, b T) bool) []T {
+	cpy := make([]T, 0, len(a))
+
+	for _, v := range a {
+		if ContainsCompare(cpy, v, equal) {
```

Comment:
Why the repeated iteration? Can you not just keep a `map[T]struct{}` around? Or are you trying to avoid the `comparable` constraint?

Reply:
I am trying to avoid the `comparable` constraint because the RBAC uses `RoleIdentifier` structs which are not comparable :cry:.

Reply:
Gotcha. We probably want to be careful that we don't use this for too large a slice then.

Reply:
Yea, atm the slices are in the order of 5-10 elements